vscode-dotnet-runtime-library/src/EventStream/EventStreamRegistration.ts

```diff
@@ -0,0 +1,55 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { IExtensionConfiguration } from '../IExtensionContext';
+import { EventStream } from './EventStream';
+import { IEventStreamObserver } from './IEventStreamObserver';
+import { LoggingObserver } from './LoggingObserver';
+import { OutputChannelObserver } from './OutputChannelObserver';
+import { StatusBarObserver } from './StatusBarObserver';
+import { ITelemetryReporter, TelemetryObserver } from './TelemetryObserver';
+
+export interface IEventStreamContext {
+    displayChannelName: string;
+    logPath: string;
+    extensionId: string;
+    enableTelemetry: boolean;
+    telemetryReporter: ITelemetryReporter | undefined;
+}
+
+export function registerEventStream(context: IEventStreamContext): [EventStream, vscode.OutputChannel, LoggingObserver, IEventStreamObserver[]] {
+    const outputChannel = vscode.window.createOutputChannel(context.displayChannelName);
+    if (!fs.existsSync(context.logPath)) {
+        fs.mkdirSync(context.logPath);
+    }
+    const logFile = path.join(context.logPath, `DotNetAcquisition-${context.extensionId}-${ new Date().getTime() }.txt`);
+    const loggingObserver = new LoggingObserver(logFile);
```

Comment:
nit side note, looked into the logging observer and you currently build out a massive string to represent the log file that will eventually be written. Would be more performant to just store each log entry in an array and then call "WriteLine" effetcively on each of those once at "reportIssue" time.

Reply:
I think we already do this, the log only gets written to that file on dispose, which is called before reportIssue and on vscode close: https://github.com/dotnet/vscode-dotnet-runtime/blob/master/vscode-dotnet-runtime-library/src/EventStream/LoggingObserver.ts#L27

Reply:
Yes but it's still generating the "this.log" string behind the covers that sits in memory and is massive 😄